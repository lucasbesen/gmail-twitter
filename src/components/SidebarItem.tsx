import React from 'react';
import styled from 'styled-components';

const Wrapper = styled.div`
  background-color: #fce8e6;
  display: flex;
  height: 30px;
  margin-right: 15px;
  border-top-right-radius: 15px;
  border-bottom-right-radius: 15px;
  align-items: center;
  padding-left: 10px;
`;

const Text = styled.span`
  color: #d93025;
  font-weight: 600;
  margin-left: 20px;
`;

const Icon = styled.img.attrs({
  src: 'https://www.gstatic.com/images/icons/material/system/2x/inbox_gm_googlered600_20dp.png',
})`
  height: 20px;
`;

const SidebarItem: React.FC = () => (
  <Wrapper>
    <Icon />
    <Text>Inbox</Text>
  </Wrapper>
);

export default SidebarItem;
