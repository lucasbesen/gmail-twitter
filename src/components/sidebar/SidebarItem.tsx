import React from 'react';
import styled from 'styled-components';

interface StyledProps {
  readonly isActive: boolean;
}

const SIDEBAR_ITEM_TYPE = {
  INBOX: 'INBOX',
  STARRED: 'STARRED',
  SNOOZED: 'SNOOZED',
};

const Wrapper = styled.div<StyledProps>`
  background-color: ${props => props.isActive ? '#fce8e6' : '#fff'};
  display: flex;
  height: 30px;
  margin-right: 15px;
  border-top-right-radius: 15px;
  border-bottom-right-radius: 15px;
  align-items: center;
  padding-left: 10px;
`;

const Text = styled.span<StyledProps>`
  color: ${props => props.isActive ? '#d93025' : '#d3d3d3'};
  font-weight: 600;
  margin-left: 20px;
`;

const Icon = styled.img.attrs({
  src: 'https://www.gstatic.com/images/icons/material/system/2x/inbox_gm_googlered600_20dp.png',
})`
  height: 20px;
`;

type Props = {
  isActive?: boolean;
  type?: string;
}

const SidebarItem: React.FC<Props> = ({ isActive = false, type }) => (
  <Wrapper isActive={isActive}>
    <Icon />
    <Text isActive={isActive}>Inbox</Text>
  </Wrapper>
);

export default SidebarItem;
