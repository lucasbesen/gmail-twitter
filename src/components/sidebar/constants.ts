export const SIDEBAR_ITEM_TYPE = {
  INBOX: 'INBOX',
  STARRED: 'STARRED',
  SNOOZED: 'SNOOZED',
};
