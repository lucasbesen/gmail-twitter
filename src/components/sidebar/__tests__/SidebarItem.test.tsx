import React from 'react';
import renderer from 'react-test-renderer';

import { SIDEBAR_ITEM_TYPE } from '../constants';
import SidebarItem from '../SidebarItem';

it('should render SidebarItem active', () => {
  const component = renderer.create(
    <SidebarItem isActive={true} type={SIDEBAR_ITEM_TYPE.INBOX}>
      Inbox
    </SidebarItem>,
  );
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});

it('should render SidebarItem snoozed', () => {
  const component = renderer.create(<SidebarItem type={SIDEBAR_ITEM_TYPE.SNOOZED}>Snoozed</SidebarItem>);
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});

it('should render SidebarItem starred', () => {
  const component = renderer.create(<SidebarItem type={SIDEBAR_ITEM_TYPE.STARRED}>Starred</SidebarItem>);
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
