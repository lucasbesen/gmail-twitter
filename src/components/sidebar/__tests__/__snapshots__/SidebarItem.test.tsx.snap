// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render SidebarItem active 1`] = `
<div
  className="sc-iwsKbI jlfGwY"
>
  <svg
    aria-hidden="true"
    className="sc-dnqmqq kyoGqk StyledIconBase-sc-bdy9j4 jbPeIR"
    fill="currentColor"
    focusable="false"
    viewBox="0 0 14 16"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M14 9l-1.13-7.14c-.08-.48-.5-.86-1-.86H2.13c-.5 0-.92.38-1 .86L0 9v5c0 .55.45 1 1 1h12c.55 0 1-.45 1-1V9zm-3.28.55l-.44.89c-.17.34-.52.56-.91.56H4.61c-.38 0-.72-.22-.89-.55l-.44-.91c-.17-.33-.52-.55-.89-.55H1l1-7h10l1 7h-1.38c-.39 0-.73.22-.91.55l.01.01z"
      fillRule="evenodd"
    />
  </svg>
  <span
    className="sc-gZMcBi grJdlT"
  >
    Inbox
  </span>
</div>
`;

exports[`should render SidebarItem snoozed 1`] = `
<div
  className="sc-iwsKbI cPBdKg"
>
  <svg
    aria-hidden="true"
    className="sc-htoDjs eYbtof StyledIconBase-sc-bdy9j4 jbPeIR"
    fill="currentColor"
    focusable="false"
    viewBox="0 0 16 16"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M8 0a8 8 0 100 16A8 8 0 008 0zm2.293 11.707L7 8.414V4h2v3.586l2.707 2.707-1.414 1.414z"
    />
  </svg>
  <span
    className="sc-gZMcBi vpeEL"
  >
    Snoozed
  </span>
</div>
`;

exports[`should render SidebarItem starred 1`] = `
<div
  className="sc-iwsKbI cPBdKg"
>
  <svg
    aria-hidden="true"
    className="sc-gzVnrw bdBlBB StyledIconBase-sc-bdy9j4 jbPeIR"
    fill="currentColor"
    focusable="false"
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
    />
  </svg>
  <span
    className="sc-gZMcBi vpeEL"
  >
    Starred
  </span>
</div>
`;
